src/serialize.h

```diff
@@ -371,38 +372,73 @@ template<typename Stream, VarIntMode Mode, typename I>
 void WriteVarInt(Stream& os, I n)
 {
     CheckVarIntMode<Mode, I>();
-    unsigned char tmp[(sizeof(n)*8+6)/7];
-    int len=0;
-    while(true) {
-        tmp[len] = (n & 0x7F) | (len ? 0x80 : 0x00);
-        if (n <= 0x7F)
+    switch(Mode) {
+        case VarIntMode::BITSTREAM: {
+            unsigned char size = (sizeof(n)*8)-std::__countl_zero(n);
```

Comment:
In ec2b2ea56c06bf72e53b0ecfb4065ccf2bd09cb0:

Should add spaces around `*` and `-` and `/` and `%` throughout this code.

Reply:
I realize that the existing code is poorly formatted. We should do better with new code.

Reply:
Also, I'm very confused by this line. What is `n`? The stream of data? It looks like you're truncating any leading zeroes? Why is this safe?

Reply:
Reverted changes, Thanks!